CRM/Civicase/Service/CaseTypeCategoryOnCreateHandler.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+use CRM_Civicase_Service_CaseTypeCategoryBaseEventHandler as CaseTypeCategoryBaseEventHandler;
+use CRM_Civicase_Service_CaseTypeCategoryProcessor as CaseTypeCategoryProcessor;
+
+/**
+ * Class CRM_Civicase_Service_CaseTypeCategoryOnCreateHandler.
+ */
+class CRM_Civicase_Service_CaseTypeCategoryOnCreateHandler extends CaseTypeCategoryBaseEventHandler {
+
+  /**
+   * {@inheritDoc}
+   */
+  public function run($op, $objectName, $objectId, &$objectRef) {
```

Comment:
This class is not supposed to accept parameters for a hook. It is a service and should be usable by hooks and by any other class/function in civicase that needs to trigger a post create event for a case type category.